From 37eab1ae32429ca57545589af206ac77ad848f15 Mon Sep 17 00:00:00 2001
From: haiqiang yao <haiqiangx.yao@intel.com>
Date: Wed, 9 Mar 2016 11:03:17 +0800
Subject: [PATCH 1/5] atomisp.h:Add new parameter bnlm for isp2.7

this patch will add new parameter bnlm for isp2.7
Depends on fw patch: https://icggerrit.ir.intel.com/#/c/107991/1

Change-Id: I40a4d0703848f94e8cdbee168824e73f801ef81c
Orig-Tracked-On: https://jira01.devtools.intel.com/browse/OAM-23440
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-25608
Signed-off-by: haiqiang yao <haiqiangx.yao@intel.com>
Reviewed-on: https://android.intel.com:443/483548
---
 include/linux/atomisp.h | 34 +++++++++++++++++++++++++++++++++-
 1 file changed, 33 insertions(+), 1 deletion(-)

diff --git a/include/linux/atomisp.h b/include/linux/atomisp.h
index 1088548..bd20531 100644
--- a/include/linux/atomisp.h
+++ b/include/linux/atomisp.h
@@ -662,6 +662,38 @@ struct atomisp_xnr3_0_11_config {
 	int32_t weight_v1;
 };
 
+struct atomisp_bnlm_config  {
+	bool rad_enable;
+	int32_t rad_x_origin;
+	int32_t rad_y_origin;
+	int32_t mu_root_lut_thr[15];
+	int32_t mu_root_lut_val[16];
+	int32_t sad_norm_lut_thr[15];
+	int32_t sad_norm_lut_val[16];
+	int32_t sig_rad_lut_thr[15];
+	int32_t sig_rad_lut_val[16];
+	int32_t sig_detail_lut_thr[15];
+	int32_t sig_detail_lut_val[16];
+	int32_t rad_pow_lut_thr[15];
+	int32_t rad_pow_lut_val[16];
+	int32_t exp_coeff_a;
+	uint32_t exp_coeff_b;
+	int32_t exp_coeff_c;
+	uint32_t exp_exponent;
+	int32_t nl_0_lut_thr[15];
+	int32_t nl_0_lut_val[16];
+	int32_t nl_1_lut_thr[15];
+	int32_t nl_1_lut_val[16];
+	int32_t nl_2_lut_thr[15];
+	int32_t nl_2_lut_val[16];
+	int32_t nl_3_lut_thr[15];
+	int32_t nl_3_lut_val[16];
+	int32_t nl_th[3];
+	int32_t match_quality_max_idx[4];
+	int32_t avg_min_th;
+	int32_t max_min_th;
+};
+
 struct atomisp_parameters {
 	struct atomisp_wb_config   *wb_config;  /* White Balance config */
 	struct atomisp_cc_config   *cc_config;  /* Color Correction config */
@@ -713,7 +745,7 @@ struct atomisp_parameters {
 	struct atomisp_macc1_5_config *macc1_5_config;
 	struct atomisp_macc1_5_table *macc1_5_table;
 	struct atomisp_xnr3_0_11_config *xnr3_0_11_config;
-
+	struct atomisp_bnlm_config  *bnlm_config;
 	void	*lin_2500_config;       /* Skylake: Linearization config */
 	void	*obgrid_2500_config;    /* Skylake: OBGRID config */
 	void	*bnr_2500_config;       /* Skylake: bayer denoise config */
-- 
1.9.1

